src/sentry/api/endpoints/organization_events_timeseries.py

```diff
@@ -0,0 +1,371 @@
+            except ValueError:
+                raise ParseError(detail="topEvents must be an integer")
+            if top_events > MAX_TOP_EVENTS:
+                raise ParseError(detail=f"Can only get up to {MAX_TOP_EVENTS} top events")
+            elif top_events <= 0:
+                raise ParseError(detail="topEvents needs to be at least 1")
+
+            return top_events
+        else:
+            return 0
+
+    def get_comparison_delta(self, request: Request) -> timedelta | None:
+        if "comparisonDelta" in request.GET:
+            try:
+                return timedelta(seconds=int(request.GET["comparisonDelta"]))
+            except ValueError:
+                raise ParseError(detail="comparisonDelta must be an integer")
+        else:
+            return None
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        with sentry_sdk.start_span(op="discover.endpoint", name="filter_params") as span:
+            span.set_data("organization", organization)
+
+            top_events = self.get_top_events(request)
+            comparison_delta = self.get_comparison_delta(request)
+
+            dataset = self.get_dataset(request)
+            # Add more here until top events is supported on all the datasets
+            if top_events > 0:
+                if dataset not in TOP_EVENTS_DATASETS:
+                    raise ParseError(detail=f"{dataset} doesn't support topEvents yet")
+
+            metrics_enhanced = dataset in {metrics_performance, metrics_enhanced_performance}
+            use_rpc = dataset in {spans_indexed, ourlogs, uptime_checks}
+
+            sentry_sdk.set_tag("performance.metrics_enhanced", metrics_enhanced)
+            try:
+                snuba_params = self.get_snuba_params(
+                    # old events-stats had global_check on False for v1, trying it off to see if that works for our
+                    # new usage
+                    request,
+                    organization,
+                    check_global_views=True,
+                )
+            except NoProjects:
+                return Response([], status=200)
+
+        try:
+            self.validate_comparison_delta(comparison_delta, snuba_params, organization)
+            rollup = self.get_rollup(request, snuba_params, top_events, use_rpc)
+            axes = request.GET.getlist("yAxis", ["count()"])
+
+            events_stats = self.get_event_stats(
+                request,
+                organization,
+                top_events,
+                dataset,
+                axes,
+                request.GET.get("query", ""),
+                snuba_params,
+                rollup,
+                comparison_delta,
+            )
+            return Response(
+                self.serialize_stats_data(events_stats, axes, snuba_params, rollup, dataset),
+                status=200,
+            )
+        except ValidationError:
```

Comment:
i'm curious if theres a particular function call that returns a `ValidationError`, and if this try/except block could be narrowed